rfcs/0009-separate-wal.md

```diff
@@ -0,0 +1,99 @@
+# Separate Object Store for WAL
+
+Status: Draft
+
+Authors:
+
+* [Sergei Sitnikov](https://github.com/taburet)
+
+## Motivation
+
+Sequential writes are a common pattern. Combined with `await_durable` set to
+`true`, which is the default, WAL performance is the main limiting factor for
+the sequential writes throughput. Even if `flush_interval` is set to an extremely
+low value, it can't overcome the inherent latency of the underlying object store.
+
+The obvious solution is switching to a lower latency object store; for instance,
+from S3 to S3 Express One Zone. But lower latency stores usually come with a cost,
+they might be significantly more expensive or might be limited in features. Using
+such stores for the main store containing the entire database might be undesirable.
+
+## Goals
+
+- Support an optional separate object store dedicated specifically for WAL.
+- Provide a user-facing configuration mechanism.
+- Avoid significant refactorings.
+
+## Public API
+
+The only change on the public API level is configuration:
+
+```rust
+pub struct DbOptions {
```

Comment:
Previously, we only need path to the DB to open a DB.
A DB with two different object store locations is only correct when the WAL location is read as well, all the time.
Should we persist this state in the manifest?


What if the DB was opened with a different wal object store first time, and then on restart, it used the same one as manifest?
It would miss reading wal. While this is a user error, the api allows for this to happen.

Reply:
I see a similar comment here, https://github.com/slatedb/slatedb/pull/533#pullrequestreview-2733846125

Reply:
> What if the DB was opened with a different wal object store first time, and then on restart, it used the same one as manifest?
> It would miss reading wal. While this is a user error, the api allows for this to happen.

https://github.com/slatedb/slatedb/pull/533#discussion_r2031541072